### PR TITLE
Add Common Event Triggers to Crafting Items

### LINKED ISSUE #812: Ability to add a crafting sound to a crafting table
**Is your feature request related to a problem? Please describe.**
Crafting feels a little strange without auditory feedback.

**Describe the solution you'd like**
Being able to play a sound for each item that's crafted would be cool. I imagine a simple first-step implementation is just to allow the user to set a sound on a crafting table - this sound would play for the client every time it crafts an item while within that crafting table.

### PR DESCRIPTION
**Pull request by AisenGaming**
Resolves #812 & #763

### CODE REVIEW

File: Intersect.Editor/Forms/Editors/frmCrafts.cs [hunk 1]
```diff
@@ -41,6 +42,9 @@ public FrmCrafts()
             cmbIngredient.Items.Clear();
             cmbIngredient.Items.Add(Strings.General.none);
             cmbIngredient.Items.AddRange(ItemBase.Names);
+            cmbEvent.Items.Clear();
```
Comment (Cheshire92) [inline Intersect.Editor/Forms/Editors/frmCrafts.cs:45]: Localisation please.

**Review by Cheshire92 — CHANGES_REQUESTED** [review]

File: Intersect.Editor/Forms/Editors/frmCrafts.cs (hunk 1 above)
Comment (AisenGaming) [reply Intersect.Editor/Forms/Editors/frmCrafts.cs:45]: Done

File: Intersect.Editor/Localization/Strings.cs [hunk 2]
```diff
@@ -1141,6 +1141,8 @@ public struct CraftsEditor
 
             public static LocalizedString undotitle = @"Undo Changes";
 
+            public static LocalizedString commonevent = @"Common Event:";
```
Comment (Cheshire92) [inline Intersect.Editor/Localization/Strings.cs:1144]: Casing and JSON properties are required for new additions, see other recently added ones (ie Not null and null value handling)

Something we often forget but probably shouldn't.

Comment (jcsnider) [reply Intersect.Editor/Localization/Strings.cs:1144]: I don't think we use NotNull anymore. Think Panda removed all those when we removes the Jetbrains libraries?

**Comment by Cheshire92** [comment]
I'm also debating on whether this should be triggerable here or as an actual event trigger you can get through the autorun triggers.. maybe both?

@jcsnider thoughts?

File: Intersect.Editor/Localization/Strings.cs (hunk 2 above)
Comment (Cheshire92) [reply Intersect.Editor/Localization/Strings.cs:1144]: Is it gone from the others? 
Might be misremembering that.

**Comment by jcsnider** [comment]
I like this. More inline with how others work, and using common event triggers would be a larger performance hit due to and looping/iteration to see what events to trigger.

Both would be fine, but this is a good start. If we do the common event triggers as well it can be a separate PR.

**Comment by Cheshire92** [comment]
Fair. This is more versatile and less of a performance hit.
Just making sure.

File: Intersect.Editor/Localization/Strings.cs (hunk 2 above)
Comment (AisenGaming) [reply Intersect.Editor/Localization/Strings.cs:1144]: Not entirely sure I follow here on what you want me to change/add. The casing is the same as the other things around it.

**Comment by jcsnider** [comment]
Fixed the merge conflict and added the null value handler.

**Review by Cheshire92 — APPROVED** [review]

File: Intersect.Editor/Forms/Editors/frmCrafts.cs [hunk 3]
```diff
@@ -549,6 +554,10 @@ private void txtSearch_Click(object sender, EventArgs e)
 
         #endregion
 
+        private void darkComboBox1_SelectedIndexChanged(object sender, EventArgs e)
```
Comment (jcsnider) [inline Intersect.Editor/Forms/Editors/frmCrafts.cs:557]: Rename this function to cmbEvent_SelectedIndexChanged and move to line 454 above
```
#region "Item List - Folders, Searching, Sorting, Etc"
```

**Review by jcsnider — CHANGES_REQUESTED** [review]

File: Intersect.Editor/Forms/Editors/frmCrafts.cs (hunk 3 above)
Comment (jcsnider) [reply Intersect.Editor/Forms/Editors/frmCrafts.cs:557]: And then make sure it still compiles/runs and it should be good to go.

Comment (AisenGaming) [reply Intersect.Editor/Forms/Editors/frmCrafts.cs]: Done

**Review by jcsnider — APPROVED** [review]

**Pull request merged**